mlflow/entities/trace.py

```diff
@@ -21,12 +21,13 @@ class Trace(_MlflowObject):
     info: TraceInfo
     data: TraceData
 
+    def to_dict(self, dump_events=False) -> Dict[str, Any]:
```

Comment:
@serena-ruan Can we implement `from_dict()` while we're here to support log unpacking use cases? cc @sunishsheth2009

Reply:
@serena-ruan Why not always dump events? IMO we should log as much information as possible

Reply:
Because events attributes values are AttributeValue = Union[
    str,
    bool,
    int,
    float,
    Sequence[str],
    Sequence[bool],
    Sequence[int],
    Sequence[float],
]
It doesn't need to be dumped in most of the cases. While if we need to write to delta table we should dump to string for consistency on spark structureType --> but if we're going to dump whole trace into string then we also don't need to dump events here, WDYT?

Reply:
Will drop dump_events